doc/user/layouts/partials/create-source/envelope/upsert/details.html

```diff
@@ -2,7 +2,13 @@
 
 Specifying `ENVELOPE UPSERT` creates a source that supports Kafka's standard
 key-value convention, and supports inserts, updates, and deletes within
-Materialize. The source is also compatible with Kafka's log-compaction feature.
+Materialize. The source is also compatible with Kafka's log-compaction feature,
+and can be useful for users who want to compact
+[CDC sources](https://materialize.com/docs/third-party/debezium/).
+
+While the upsert envelope format is more expressive, it comes with the tradeoffs
+of being slower and more memory intensive to ingest, because processing this format
+requires maintaining state proportional to the number of unique keys.
```

Comment:
Technically, this is saying it is more memory intensive to ingest the upsert envelope format itself, which I don't think is what you mean? I'm also not clear on what it's more expressive or slower *than*.

Maybe this?

> While the upsert envelope format is more expressive than the default (append-only) envelope format, it makes data ingestion slower and more memory intensive, because processing this format requires maintaining state proportional to the number of unique keys.

Reply:
thanks!